swap/src/monero/wallet.rs

```diff
@@ -129,15 +172,17 @@ impl WatchForTransfer for Wallet {
         }
 
         let address = Address::standard(self.network, public_spend_key, public_view_key.into());
-        let wallet = self.inner.clone();
```

Comment:
@thomaseizinger note that I changed the way the mutex is locked here (within each retry rather than locking the wallet beforehand) because I think it will behave differently and the wallet should not be locked for the complete run of the function. However, I am not totally sure the behavior really changed. Would be nice if we can think about this together tomorrow.